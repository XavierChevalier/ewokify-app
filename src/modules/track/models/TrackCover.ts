import { z } from 'zod'
import { TrackBands, TrackTitle } from '@/modules/track/models/Track'

export const TrackCover = z.string().url()
export type TrackCover = z.infer<typeof TrackCover>

export const TrackOfTrackCover = z.strictObject({
  cover: TrackCover,
  title: TrackTitle,
  bands: TrackBands,
})
export type TrackOfTrackCover = z.infer<typeof TrackOfTrackCover>
