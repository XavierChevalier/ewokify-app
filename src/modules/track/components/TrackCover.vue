<script setup lang="ts">
  import { TrackOfTrackCover } from '@/modules/track/models/TrackCover'
  import { computed, PropType } from 'vue'
  import { isPropertyValid } from '@/tools/component-properties/PropertyValidator'

  const props = defineProps({
    size: {
      type: Number,
      required: false,
      default: 4,
    },
    track: {
      type: Object as PropType<TrackOfTrackCover>,
      required: true,
      validator: isPropertyValid(TrackOfTrackCover),
    },
  })

  const alt = computed(() => {
    const bands = props.track.bands.join(', ')

    return `The track cover of ${props.track.title} â€¢ ${bands}`
  })
  const size = computed(() => `${props.size}rem`)
</script>

<template>
  <img
    class="rounded"
    :src="track.cover"
    :alt="alt"
    :style="{ width: size, height: size }"
  />
</template>
