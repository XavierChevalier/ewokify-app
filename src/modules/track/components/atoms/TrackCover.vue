<script lang="ts">
  import { defineComponent } from 'vue'
  import { SizeMixin } from '@/mixins/SizeMixin'

  export default defineComponent({
    mixins: [
      SizeMixin({
        sm: 'w-9 h-9',
        base: 'w-16 h-16',
      }),
    ],
  })
</script>

<script setup lang="ts">
  import { TrackOfTrackCover } from '@/modules/track/models/TrackCover'
  import { computed, PropType } from 'vue'
  import { isPropertyValid } from '@/tools/component-properties/PropertyValidator'

  const props = defineProps({
    track: {
      type: Object as PropType<TrackOfTrackCover>,
      required: true,
      validator: isPropertyValid(TrackOfTrackCover),
    },
  })

  const alt = computed(() => {
    const bands = props.track.bands.join(', ')

    return `The track cover of ${props.track.title} â€¢ ${bands}`
  })
</script>

<template>
  <img class="rounded" :class="sizeClass" :src="track.cover" :alt="alt" />
</template>
