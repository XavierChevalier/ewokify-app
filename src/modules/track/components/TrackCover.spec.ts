import { generateStorybookSnapshotTests } from '@/tests/tools/StorybookSnapshot'
import { Default } from '@/modules/track/components/TrackCover.stories'

describe('TrackCover', () => {
  generateStorybookSnapshotTests({ Default })
})
