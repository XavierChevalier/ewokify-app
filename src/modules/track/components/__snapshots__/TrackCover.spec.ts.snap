// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackCover [snapshot] should render the "Default" component correctly 1`] = `
<img
  alt="The track cover of G.O.A.T. • Polyphia"
  class="rounded w-16 h-16"
  src="https://picsum.photos/50"
/>
`;

exports[`TrackCover [snapshot] should render the "WithSizeOf6" component correctly 1`] = `
<img
  alt="The track cover of G.O.A.T. • Polyphia"
  class="rounded"
  src="https://picsum.photos/50"
  style="width: 6rem; height: 6rem;"
/>
`;
